### PR TITLE
Vendored ccm

### PR DESCRIPTION
**Pull request by nckturner**
Creates aws ccm binary, with the cloud provider specific code vendored in-tree code.  It should eventually migrate to this repository, but this will help us get started with e2e testing.

### CODE REVIEW

**Comment by gnufied** [comment]
Discussed this in meeting today. Lets merge this for now.

/lgtm

File: cmd/aws-cloud-controller-manager/main.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apiserver/pkg/server/healthz"
+	"k8s.io/apiserver/pkg/util/logs"
+	"k8s.io/kubernetes/cmd/cloud-controller-manager/app"
+	"k8s.io/kubernetes/cmd/cloud-controller-manager/app/options"
+	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
+	_ "k8s.io/kubernetes/pkg/cloudprovider/providers/aws"
```
Comment (andrewsykim) [inline cmd/aws-cloud-controller-manager/main.go]: This is how all out-of-tree providers are doing it now, but in the near future, this import path may change depending on [this proposal](https://docs.google.com/document/d/1YqsPzeG10E59vea5StKFM6W_xZq034eTaf1n649JQvs/edit).

Comment (nckturner) [reply cmd/aws-cloud-controller-manager/main.go]: Yup, this was just meant to get this repository building a binary, and keep the source of truth in-tree.  Once the shared, generic cloudprovider code moves to staging, these imports will reflect that.  And once the aws code moves into *this repository*, these imports will reflect that too.

**Comment by justinsb** [comment]
We also discussed this in sig-aws on Friday (June 1st 2018).  Consensus was to proceed with this PR, in that it allows progress and doesn't really hurt us.  It might be possible to do more in-tree with bazel restrictions etc, but that shouldn't stop us proceeding down this path also.

**Comment by jsafrane** [comment]
Do we have merge bot running here?

**Comment by jsafrane** [comment]
We do :-)

**Pull request merged**